File: 27-remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int ptr=0;
        int k=nums.length;
        while (ptr<k)
        {
            if (nums[ptr]==val){
                int temp=nums[k-1];
                nums[k-1]=nums[ptr];
                nums[ptr]=temp;
                k--;
            }else{
                ptr++;
            }
        }
        return k;
    }
}

File: 27-remove-element/remove-element.go
func removeElement(nums []int, val int) int {
	ptr_start := 0
	ptr_end := len(nums) - 1

	for ptr_start <= ptr_end {
		if nums[ptr_start] == val {
			nums[ptr_start], nums[ptr_end] = nums[ptr_end], nums[ptr_start]
			ptr_end--
		} else {
			ptr_start++
		}
	}
	return ptr_start
}